File: migration_plan/update_progress.js
const fs = require('fs');
const path = require('path');
const csv = require('csv-parse/sync');
const { stringify } = require('csv-stringify/sync');

const componentsToMark = [
    // Shared Layouts
    'MobileLayout',
    'MobileNav',
    'MobileAIWizardLayout',
    // Auth
    'Login', // Assuming Login.jsx isn't in parsing list but if it was
    // Home
    'MobileHome',
    'MobileHeroDashboardPreview',
    // UI Components
    'PrimaryButton',
    'AppInput',
    'MobileMarkdownRenderer', // We ported this
    'MobileStoryDisplay', // We ported this to StoryDisplay
    // AI Gateway
    'MobileAIGateway',
    'MobileAIGenerator',
    'MobileGenStory',
    'MobileGenDialogue',
    'MobileGenArticle'
];
const csvPath = 'e:\\vocab_web\\migration_plan\\MIGRATION_COMPONENTS_DEEP.csv';

try {
    // Read file
    const input = fs.readFileSync(csvPath, 'utf8');

    // Parse CSV
    const records = csv.parse(input, {
        columns: true,
        skip_empty_lines: true
    });

    // Update records
    let updatedCount = 0;
    for (const record of records) {
        // Add columns if missing (csv-parse treats them as keys)
        if (!record['Is Migrated']) record['Is Migrated'] = 'FALSE';
        if (!record['Is Tested']) record['Is Tested'] = 'FALSE';

        if (componentsToMark.includes(record['Component Name'])) {
            record['Is Migrated'] = 'TRUE';
            record['Is Tested'] = 'TRUE';
            updatedCount++;
            console.log(`Marked ${record['Component Name']} as Migrated & Tested`);
        }
    }

    // Write back
    const output = stringify(records, {
        header: true
    });

    fs.writeFileSync(csvPath, output);
    console.log(`Updated ${updatedCount} components.`);

} catch (err) {
    console.error('Error:', err);
}

File: migration_plan/update_progress.py
import csv
import sys
import os

def update_csv(file_path, completed_components):
    # Read all rows
    rows = []
    with open(file_path, 'r', newline='', encoding='utf-8') as f:
        reader = csv.reader(f)
        header = next(reader)
        rows = list(reader)

    # Add new columns if they don't exist
    new_columns = ['Is Migrated', 'Is Tested']
    column_indices = {}
    
    for col in new_columns:
        if col not in header:
            header.append(col)
            # Add empty values to existing rows
            for row in rows:
                row.append('FALSE')
        
    # Map column indices
    for i, col in enumerate(header):
        column_indices[col] = i

    # Update rows
    migrated_idx = column_indices['Is Migrated']
    tested_idx = column_indices['Is Tested']
    name_idx = column_indices['Component Name']

    updated_count = 0
    for row in rows:
        name = row[name_idx]
        if name in completed_components:
            row[migrated_idx] = 'TRUE'
            row[tested_idx] = 'TRUE'
            updated_count += 1
            print(f"Marked {name} as Migrated & Tested")

    # Write back
    with open(file_path, 'w', newline='', encoding='utf-8') as f:
        writer = csv.writer(f)
        writer.writerow(header)
        writer.writerows(rows)

    print(f"Updated {updated_count} components.")

if __name__ == "__main__":
    csv_path = r'e:\vocab_web\migration_plan\MIGRATION_COMPONENTS_DEEP.csv'
    components_to_mark = ['MobileLayout', 'MobileNav', 'MobileAIWizardLayout']
    update_csv(csv_path, components_to_mark)
